src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -580,6 +580,40 @@ public ExpressionStringPojo getConceptNormalForm(
 
 		return new ExpressionStringPojo(expression.toString(includeTerms));
 	}
+	
+	@PostMapping(value = "/{branch}/concepts/donate")
+	@PreAuthorize("hasPermission('AUTHOR', #branch)")
+	@JsonView(value = View.Component.class)
+	public Collection<ConceptMini> donateConcepts(
+			@ApiParam(value = "Destination branch where the concepts are copied to.")
+			@PathVariable String branch,
+			@ApiParam(value = "Source branch where the concepts are selected from.")
+			@RequestParam String sourceBranch,
+			@ApiParam(value = "ECL expression for selecting concepts to copy between the branches.")
+			@RequestParam String ecl,
+			@ApiParam(value = "Include dependant components. Defaults to false.")
+			@RequestParam(required = false, defaultValue = "false") boolean includeDependencies) throws ServiceException {
+		String sourceBranchPath = BranchPathUriUtil.decodePath(sourceBranch);
```

Comment:
@astro-snail It would be good if we add checks to make sure the source and destination are branches from different modules to prevent this being misused.